core/src/main/java/com/linecorp/armeria/server/ServerCacheControlBuilder.java

```diff
@@ -0,0 +1,185 @@
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server;
+
+import java.time.Duration;
+
+import javax.annotation.Nullable;
+
+import com.linecorp.armeria.client.ClientCacheControlBuilder;
+import com.linecorp.armeria.common.CacheControlBuilder;
+
+/**
+ * Creates a new {@link ServerCacheControl} using the builder pattern.
+ *
+ * <pre>{@code
+ * ServerCacheControl cacheControl =
+ *     new ServerCacheControlBuilder()
+ *         .noCache()
+ *         .noStore()
+ *         .mustRevalidate()
+ *         .build();
+ * }</pre>
+ *
+ * @see ClientCacheControlBuilder
+ * @see <a href="https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/http-caching">HTTP Caching (Google)</a>
+ * @see <a href="https://developer.mozilla.org/docs/Web/HTTP/Headers/Cache-Control">Cache-Control (MDN)</a>
+ */
+public final class ServerCacheControlBuilder extends CacheControlBuilder<ServerCacheControlBuilder> {
+
+    private boolean cachePublic;
+    private boolean cachePrivate;
+    private boolean cacheImmutable;
+    private boolean mustRevalidate;
+    private boolean proxyRevalidate;
+    private long sMaxAgeSeconds = -1;
+
+    /**
+     * Creates a new builder with all directives disabled initially.
+     */
+    public ServerCacheControlBuilder() {}
+
+    ServerCacheControlBuilder(ServerCacheControl c) {
+        super(c);
+        cachePublic = c.cachePublic();
+        cachePrivate = c.cachePrivate();
+        cacheImmutable = c.cacheImmutable();
+        mustRevalidate = c.mustRevalidate();
+        proxyRevalidate = c.proxyRevalidate();
+        sMaxAgeSeconds = c.sMaxAgeSeconds();
+    }
+
+    /**
+     * Enables the {@code "public"} directive.
+     */
+    public ServerCacheControlBuilder cachePublic() {
+        return cachePublic(true);
+    }
+
+    /**
+     * Enables or disables the {@code "public"} directive.
+     *
+     * @param cachePublic {@code true} to enable or {@code false} to disable.
+     */
+    public ServerCacheControlBuilder cachePublic(boolean cachePublic) {
+        this.cachePublic = cachePublic;
+        return this;
+    }
+
+    /**
+     * Enables the {@code "private"} directive.
+     */
+    public ServerCacheControlBuilder cachePrivate() {
+        return cachePrivate(true);
+    }
+
+    /**
+     * Enables or disables the {@code "private"} directive.
+     *
+     * @param cachePrivate {@code true} to enable or {@code false} to disable.
+     */
+    public ServerCacheControlBuilder cachePrivate(boolean cachePrivate) {
+        this.cachePrivate = cachePrivate;
```

Comment:
Just wondering if we need to check the exclusivity of `private` and `public` directives.

Reply:
Made sure `cachePrivate` has precedence over `cachePublic`, i.e. if both are set, `cachePrivate()` returns `true` and `cachePublic()` returns `false`.